src/trove/chart.js.rkt

```diff
@@ -0,0 +1,835 @@
+          (args ("self"))
+          (return ,Image)
+          (doc ("Produce an " ,Image " of the chart"))))))
+    (constr-spec
+      (name "plot-chart-window")
+      (with-members (,x-min-meth ,x-max-meth ,y-min-meth ,y-max-meth ,x-axis-meth ,y-axis-meth
+                     ,num-samples-meth)))
+    (constr-spec
+      (name "histogram-chart-window")
+      (with-members (,x-min-meth ,x-max-meth ,y-max-meth ,x-axis-meth ,y-axis-meth)))
+    (constr-spec
+      (name "bar-chart-window")
+      (with-members (,y-min-meth ,y-max-meth ,x-axis-meth ,y-axis-meth)))
+    (constr-spec
+      (name "pie-chart-window")
+      (with-members ()))
+  ))
+
+@docmodule["chart"]{
+  The Pyret Chart library. It consists of chart, plot, and data visualization tools,
+  using @link["https://developers.google.com/chart/" "Google Charts"] as a backend.
+
+  This documentation assumes that you @pyret{include} the chart library,
+  and has imported @pyret{image-structs} as @pyret{I} (@pyret{import image-structs as I}).
+
+  @;############################################################################
+  @section{Creating a DataSeries}
+
+  Given data, before you visualize them as a chart, you need to choose what @emph{type}
+  of chart do you want. For example, you could either choose to visualize the data as
+  a bar chart or as a pie chart.
+
+  The data along with the type of chart and chart-specific configuration is
+  called a @|DataSeries|. As an example,
+  you might have data about English native speakers in several countries and would like
+  to visualize them using bar chart. The data could be represented by a list of
+  strings (country names) and a list of numbers (number of English native speakers) as follows:
+
+  @pyret-block{
+countries =    [list: "US",      "India",   "Pakistan", "Philippines", "Nigeria"]
+num-speakers = [list: 251388301, 125344736, 110041604,  89800800,      79000000]
+  }
+
+  Then, you can use a @emph{chart constructor} -- here, the bar chart constructor
+  @pyret{from-list.bar-chart} -- to create a @|DataSeries|:
+
+  @pyret-block{
+a-pie-chart-series = from-list.bar-chart(countries, num-speakers)
+  }
+
+  As another example, say we have a function in Pyret @pyret{fun f(x): num-sin(2 * x) end}.
+  We can use the chart constructor @pyret{from-list.function-plot} to create a
+  @|DataSeries|. Furthermore, we can specify the color of the function by using
+  the method @pyret{color} on the @|DataSeries| to obtain a new @|DataSeries|
+  with the desired color.
+
+  @pyret-block{
+import image-structs as I
+
+fun f(x): num-sin(2 * x) end
+intermediate-series = from-list.function-plot(f)
+a-series = intermediate-series.color(I.purple)
+  }
+
+  If you prefer, you can chain methods to avoid naming the intermediate series:
+
+  @pyret-block{
+a-series = from-list.function-plot(f)
+  .color(I.purple)
+  }
+
+  @margin-note{We plan that the chart library should support the @pyret-id["Table" "tables"] inferface too.
```

Comment:
(CC: @schanzer) Good question. What do you think is the best interface for this, then?

Reply:
I don't know enough about API design to have an informed opinion here. Part of me says "why not just make that available without having to say `from-X`?" But maybe that introduces an inconsistency that'll bite us later. Could use input from @blerner and @jpolitz here. 

(And a reminder that nothing should change here on CPO till Sat night, since I'll be banging on the charts API for a few days)